src/test/java/org/jnode/fs/xfs/XfsV4FileSystemTest.java

```diff
@@ -126,4 +127,19 @@ public <T> List<T> iteratorToList(Iterator<T> iterator) {
         Spliterator<T> spliterator = Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED);
         return StreamSupport.stream(spliterator, false).collect(Collectors.toList());
     }
+
+
+    @Ignore("test data not in project, it is 10.5GB, too large to put in code.")
+    @Test
+    public void testSocket() throws Exception {
+        File testFile = FileSystemTestUtils.getTestFile("org/jnode/fs/xfs/v4/ubuntu_xfs_v4.img");
+        try (FileDevice device = new FileDevice(testFile, "r")) {
+            XfsEntry entry = DataTestUtils.getDescendantData(new XfsFileSystemType().create(device, true), "tmp", "keyring-eoplNH", "ssh");
+
+            assertThat(entry.getName(), is ("ssh"));
+            assertThat(entry.getId(), is("2084921-4"));
```

Comment:
Can add `assertThat(entry.getINode().getFileModes(), contains(FileMode.SOCKET));` to confirm that the entry is a socket.